### PR TITLE
Polygonal search

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Overview

Allows users to filter facilities search to within a polygonal boundary. 

Users are now able to draw a polygon on the map. The facility search
results are filtered to only include the facilities which are contained
within the boundaries of the polygon.

The polygon is rendered based on the boundaries filter, so it renders
correctly on page refreshes based on the query string in the URL. The
polygon is also removed when the user resets the filters, thereby
removing the boundaries.

Redux functionality has been added to support the polygonal search
feature. Given a list of points in the form of `[[lat, long]]` in the
redux state, a query string is created when searching for facilities
which filters the facilities search to only return facilities within
the polygon described by the given points.

In addition, the points listed in the URL query string are able to be
parsed in order to rehydrate the redux state on load.

The API facilities search now takes a query string in the form
`&boundaries=lat,lng&boundaries=lat,lng` etc, where each incidence of
`&boundaries=lat,lng` represents a single point of a polygon. Search
results are filtered to only include facilities for which their
location is contained within the polygon.

Connects #793 

## Demo

polygon

## Notes

I render the polygon as an SVG to avoid blocking clicks on the main map.

## Testing Instructions

* Check out this branch
* Run `vagrant ssh` and `./scripts/server`
* Click on the polygon button on the map, then click on points until you've drawn a polygon.
* The search results should be filtered to only show results within the polygon.
* You should be able to click on search results as usual.
* Refresh the page; the polygon should remain unchanged.
* Click the trashcan button and the polygon filter should be removed and the search re-run.
* Create a new polygon. 
* Click 'reset' in the search tab.
* The polygon filter should be removed. 

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Comment by jwalgran** [comment]
Looking at this now.

I see there is a conflict, and some of the commits appear to come from the recently merged #966. It would be good to rebase this branch on `develop`.

File: src/app/src/components/PolygonalSearchControl.jsx [hunk 1]
```diff
@@ -77,15 +80,16 @@ class PolygonalSearchControl extends React.Component {
         const svgRenderer = L.svg({ padding: 0.5 });
 
         return (
-            <FeatureGroup ref="drawnItems">
+            <FeatureGroup ref={this.editableLayers}>
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/app/src/components/PolygonalSearchControl.jsx]: Added note to commit message re: string refs as legacy API

**Comment by jwalgran** [comment]
Zooming to the search area does not seem to be working properly

**Comment by TaiWilkin** [comment]
Taking another look at the polygonal zoom now

**Comment by TaiWilkin** [comment]
Zooming to polygon wasn't working due to mismatch between Leaflet [lat, lng] params and GeoJSON [lng, lat]. Corrected with an explicit reformatting to a { lat, lng} object.

**Comment by designmatty** [comment]
I don't have a ton of suggestions here, but this is the general thinking:
- Move the button down below the "Country" filter 
    - (these two filters are close in their functionality so I thought it made sense to have them together.)
- Add a label  above the button

image

**Pull request merged**